### ISSUE TITLE
Not working on Kubernetes 1.25

### ISSUE THREAD

**Issue by mateuszkula1** [FairwindsOps/gemini#146]
Hello,

I was using gemini in my clusters with success so far.
Unfortunately project is not compatible with k8s in version 1.25+ and I am not able to deploy SnapshotGroup on updated clusters.
I don't see a lot on activity in the project. Is it abandoned and won't be developed anymore?

**Comment by rbren** [comment]
Hi @mateuszkula1, thanks for reaching out!

Gemini is being maintained, but not actively developed.

We'll try and get things compatible with 1.25 ASAP

**Comment by mateuszkula1** [comment]
I'm glad to hear that. To be more precise I think it could stop working even for earlier versions of k8s like 1.22. Problem is that CRD Api changed and gemini was not aligned to this.
Anyway keeping fingers crossed for fix to be delivered soon. 
Have a nice day!

**Comment by rbren** [comment]
@mateuszkula1 so far I'm unable to reproduce your issue. Can you share more about the specific error you're seeing? I'd also be curious to see:
* `kubectl version`
* `kubectl api-resources`

Does uninstalling Gemini entirely and reinstalling it work? Or does a fresh install break as well?

We do have support for `apiextensions.k8s.io/v1/CustomResourceDefinition` as well as the v1beta1 that was removed in 1.22. So it _should_ work fine.

Unfortunately it's hard to test properly in KIND (due to CSI support), so I'm going to keep digging in

**Comment by lukpep** [comment]
@rbren I will provide some context on behalf of @mateuszkula1, with whom I'm working on the subject.
Initially, we installed Gemini on the 1.21 cluster, and it worked fine - It was configured to create backups of several PVCs.
The mentioned cluster was then updated sequentially to 1.25 and Gemini still works - to this day.
A problem occurred when we tried to install it on a new cluster - 1.25 at the time and it seems to be well described in the 
https://github.com/FairwindsOps/gemini/issues/68

Mentioned workaround worked and it's to manually change CRD definition to match the newly changed PVC structure: https://gist.github.com/Aste88/0c70dea7719e06199a3b281625f9f6d1

Fresh install with objects:

```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  labels:
  name: lpe-test
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 4Gi
  storageClassName: standard-rwo
```
```
apiVersion: v1
kind: Pod
metadata:
  name: task-pv-pod
spec:
  volumes:
    - name: task-pv-storage
      persistentVolumeClaim:
        claimName: lpe-test
  containers:
    - name: task-pv-container
      image: nginx
      ports:
        - containerPort: 80
          name: "http-server"
      volumeMounts:
        - mountPath: "/usr/share/nginx/html"
          name: task-pv-storage

```

and snap (taken from docs):
```
apiVersion: gemini.fairwinds.com/v1beta1
kind: SnapshotGroup
metadata:
  name: test-volume
spec:
  persistentVolumeClaim:
    claimName: lpe-test
  schedule:
    - every: day
      keep: 14
```

results in errors in logs:

```
I0210 11:54:47.064149       1 controller.go:181] Starting SnapshotGroup controller
I0210 11:54:47.064176       1 controller.go:183] Waiting for informer caches to sync
I0210 11:54:47.164564       1 controller.go:188] Starting workers
I0210 11:54:47.164592       1 controller.go:193] Started workers
I0210 11:59:25.152981       1 groups.go:38] default/test-volume: reconciling
I0210 11:59:25.157869       1 pvc.go:48] default/lpe-test: PVC found
I0210 11:59:25.157895       1 groups.go:29] default/test-volume: updating PVC spec
W0210 11:59:25.167254       1 warnings.go:70] unknown field "spec.persistentVolumeClaim.spec.accessModes"
W0210 11:59:25.167270       1 warnings.go:70] unknown field "spec.persistentVolumeClaim.spec.resources"
W0210 11:59:25.167273       1 warnings.go:70] unknown field "spec.persistentVolumeClaim.spec.storageClassName"
W0210 11:59:25.167275       1 warnings.go:70] unknown field "spec.persistentVolumeClaim.spec.volumeMode"
W0210 11:59:25.167277       1 warnings.go:70] unknown field "spec.template.spec.source"
W0210 11:59:25.167279       1 warnings.go:70] unknown field "status"
I0210 11:59:25.171395       1 groups.go:53] default/test-volume: found 0 existing snapshots
I0210 11:59:25.171416       1 scheduler.go:91] need creation day true
I0210 11:59:25.171421       1 groups.go:59] default/test-volume: going to create 1, delete 0 snapshots
I0210 11:59:25.171426       1 groups.go:65] default/test-volume: deleted 0 snapshots
I0210 11:59:25.171433       1 snapshots.go:136] default/test-volume: creating snapshot for PVC lpe-test
I0210 11:59:25.192456       1 groups.go:71] default/test-volume: created 1 snapshots
I0210 11:59:25.192483       1 controller.go:146] default/test-volume: successfully performed backup
I0210 11:59:25.192492       1 groups.go:38] default/test-volume: reconciling
I0210 11:59:25.198298       1 pvc.go:48] default/lpe-test: PVC found
I0210 11:59:25.198318       1 groups.go:29] default/test-volume: updating PVC spec
W0210 11:59:25.204379       1 warnings.go:70] unknown field "spec.persistentVolumeClaim.spec.accessModes"
W0210 11:59:25.204397       1 warnings.go:70] unknown field "spec.persistentVolumeClaim.spec.resources"
W0210 11:59:25.204400       1 warnings.go:70] unknown field "spec.persistentVolumeClaim.spec.storageClassName"
W0210 11:59:25.204403       1 warnings.go:70] unknown field "spec.persistentVolumeClaim.spec.volumeMode"
W0210 11:59:25.204406       1 warnings.go:70] unknown field "spec.template.spec.source"
W0210 11:59:25.204409       1 warnings.go:70] unknown field "status"
I0210 11:59:25.208292       1 groups.go:53] default/test-volume: found 1 existing snapshots
I0210 11:59:25.208319       1 scheduler.go:91] need creation day false
I0210 11:59:25.208325       1 groups.go:59] default/test-volume: going to create 0, delete 0 snapshots
I0210 11:59:25.208332       1 groups.go:65] default/test-volume: deleted 0 snapshots
I0210 11:59:25.208336       1 groups.go:71] default/test-volume: created 0 snapshots
I0210 11:59:25.208345       1 controller.go:146] default/test-volume: successfully performed backup
```

this is solvable by changeind PVC template in CRD definition.

```
k version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.2", GitCommit:"5835544ca568b757a8ecae5c153f317e5736700e", GitTreeState:"clean", BuildDate:"2022-09-21T14:33:49Z", GoVersion:"go1.19.1", Compiler:"gc", Platform:"darwin/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.5-gke.1500", GitCommit:"341714971fd5f72a615e49c935c0e4d19e2599e0", GitTreeState:"clean", BuildDate:"2022-12-28T09:36:24Z", GoVersion:"go1.19.4 X:boringcrypto", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by rbren** [comment]
Ahh! That makes a lot more sense. Appreciate the info--I'll take a look

**Comment by rbren** [comment]
This is fixed in 2.0. Chart upgrade is incoming
